packages/next/src/server/lib/router-utils/setup-dev-bundler.ts

```diff
@@ -149,6 +151,14 @@ async function startWatcher(opts: SetupOpts) {
 
   const distDir = path.join(opts.dir, opts.nextConfig.distDir)
 
+  // we ensure the types directory exists on turbo
+  if (usingTypeScript && opts.turbo) {
+    const distTypesDir = path.join(distDir, 'types')
+    if (!fs.existsSync(distTypesDir)) {
+      await mkdir(distTypesDir, { recursive: true })
+    }
+  }
```

Comment:
@huozhi Since Turbopack don't generate `types` dir like our current Webpack plugin does, we explicitly check and create one here (so we can ensure on the further `typed...` features)